### DISCUSSION TITLE
Any plans to support smart TVs and consoles?

### DISCUSSION THREAD

**Discussion by BProg** [tauri-apps/wry#285] [Ideas]
I know some consoles have closed SDK and someone would need developer account in order to get access to SDKs, but supporting consoles and smartTVs would be a great step forward to allow developing TV apps, not only desktop and mobile

**Comment by shirakaba** [comment]
Taking stock of a few platforms and whether they could translate well to wry:

* ✅ HbbTV: basically just a web app.
* ❌ tvOS: basically an iOS/macOS app, but **with no WKWebView**.
* ❌ Roku: coded in BrightScript, so foreign to both web tech and mobile/desktop tech.
* ✅ Android TV: seems to support WebView, so could be coded as a web app.
* ✅ Tizen: seems to support WebView too.
* ✅ Fire TV: seems to support WebView too.

So, maybe some scope due to broad WebView support. But, speaking as a software engineer working full-time on Smart TV apps: it is a real challenge to make a cross-platform abstraction for targeting a broad range of Smart TV (and video-on-demand) platforms. In particular, handling the differences in spatial navigation when targeting iOS (touch-based), tvOS (native spatial navigation engine), macOS (mouse- and keyboard-based), and web-based platforms (mouse- and keyboard-based, and/or TV remote control-based). The media types supported by each of these platforms vary wildly (especially when it comes to DRM support), and there will inevitably be many environment-specific shims needed (e.g. on HbbTV apps to handle the video/broadcast object).

So even if you could _deploy_ to all of these platforms, your app code would likely be riddled with platform-specific considerations to the point that it becomes hard to work with and a native or single-target app may be preferred. But ultimately companies like Netflix, Crunchyroll, iPlayer and Funimation **do** put in the effort to tackle these issues, so the idea probably shouldn't be discounted completely. If those problems can be delegated to the community rather than being made a responsibility of the core library, it might be worth considering.

My main worry, I think though, is that it would spread development resources very thin. It's hard to master each platform (I'm not sure how much we've even looked at Android mobile yet), especially to the point of making a workflow for deploying apps on the given ecosystem (macOS notarisation appeared to be a lot of work indeed).

Worth at least asking how much interest there is in each given platform, though.
